WebContent/bank/balance.jsp

```diff
@@ -0,0 +1,132 @@
+    pageEncoding="ISO-8859-1"%>
+
+<%
+/**
+ This application is for demonstration use only. It contains known application security
+vulnerabilities that were created expressly for demonstrating the functionality of
+application security testing tools. These vulnerabilities may present risks to the
+technical environment in which the application is installed. You must delete and
+uninstall this demonstration application upon completion of the demonstration for
+which it is intended. 
+
+IBM DISCLAIMS ALL LIABILITY OF ANY KIND RESULTING FROM YOUR USE OF THE APPLICATION
+OR YOUR FAILURE TO DELETE THE APPLICATION FROM YOUR ENVIRONMENT UPON COMPLETION OF
+A DEMONSTRATION. IT IS YOUR RESPONSIBILITY TO DETERMINE IF THE PROGRAM IS APPROPRIATE
+OR SAFE FOR YOUR TECHNICAL ENVIRONMENT. NEVER INSTALL THE APPLICATION IN A PRODUCTION
+ENVIRONMENT. YOU ACKNOWLEDGE AND ACCEPT ALL RISKS ASSOCIATED WITH THE USE OF THE APPLICATION.
+
+IBM AltoroJ
+(c) Copyright IBM Corp. 2008, 2013 All Rights Reserved.
+*/
+%> 
+    
+<jsp:include page="/header.jspf"/>
+
+<div id="wrapper" style="width: 99%;">
+	<jsp:include page="membertoc.jspf"/>
+    <td valign="top" colspan="3" class="bb">
+		<%@page import="com.ibm.security.appscan.altoromutual.model.Account"%>
+		<%@page import="java.text.SimpleDateFormat"%>
+		<%@page import="java.text.NumberFormat"%>
+		<%@page import="java.text.DecimalFormat"%>
+		<%@page import="java.util.ArrayList"%><div class="fl" style="width: 99%;">
+		
+		<%
+					com.ibm.security.appscan.altoromutual.model.User user = (com.ibm.security.appscan.altoromutual.model.User)request.getSession().getAttribute("user"); 
+					ArrayList<Account> accounts = new ArrayList<Account>();
+					java.lang.String paramName = request.getParameter("acctId");
+					String accountName = paramName;
+													
+					for (Account account: user.getAccounts()){
+						
+						if (!String.valueOf(account.getAccountId()).equals(paramName))
+							accounts.add(account);
+						else {
+							accounts.add(0, account);
+							accountName = account.getAccountId() + " " + account.getAccountName();
+						}
+					}
+				%>
+		
+		<!-- To modify account information do not connect to SQL source directly.  Make all changes
+		through the admin page. -->
+		
+		<h1>Account History - <%=accountName%></h1>
+		
+		<table width="590" border="0">
+		  <tr>
+		    <td colspan=2>
+		      <table cellSpacing="0" cellPadding="1" width="100%" border="1">
+		        <tr>
+		          <th colSpan="2">
+		            Balance Detail</th></tr>
+		        <tr>
+		          <th align="left" width="80%" height="26">
+		            <form id="Form1" method="get" action="showAccount">
+			           	<select size="1" name="listAccounts" id="listAccounts">
+							<%
+								for (Account account: accounts){
+									out.println("<option value=\""+account.getAccountId()+"\">" + account.getAccountId() + " " + account.getAccountName() + "</option>");
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from `a database` flows into `println`, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

<h6>
Line 72 | <a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 592</a> | <a href="https://learn.snyk.io/lesson/xss/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 24 steps</b></summary>

`Step 1 - 3` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L281

`Step 4 - 5` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L283](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L283)

`Step 6` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L284](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L284)

`Step 7 - 9` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L287](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L287)

`Step 10` [src/com/ibm/security/appscan/altoromutual/model/User.java#L78](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/model/User.java#L78)

`Step 11 - 12` [WebContent/bank/balance.jsp#L43](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L43)

`Step 13` [WebContent/bank/balance.jsp#L45](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L45)

`Step 14` [WebContent/bank/balance.jsp#L48](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L48)

`Step 15` [WebContent/bank/balance.jsp#L46](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L46)

`Step 16 - 17` [WebContent/bank/balance.jsp#L71](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L71)

`Step 18 - 24` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/balance.jsp#L72

</details>